scipy/sparse/csgraph/_laplacian.py

```diff
@@ -34,8 +34,9 @@ def laplacian(csgraph, normed=False, return_diag=False, use_out_degree=False):
 
     Returns
     -------
-    lap : ndarray
-        The N x N laplacian matrix of graph.
+    lap : ndarray or coo_matrix
+        The N x N laplacian matrix of csgraph. It will be a numpy array (dense)
+        if the input was dense, sparse (COO format) otherwise.
```

Comment:
This is correct, but I think we should not promise to return specifically a COO matrix. If we only promise to return a sparse matrix, then we have space to optimize by using some other format later.